app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -685,9 +687,17 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
         return super.onKeyDown(keyCode, event)
     }
 
-    final override fun onKeyUp(keyCode: Int, event: KeyEvent?): Boolean {
+    final override fun onKeyUp(keyCode: Int, event: KeyEvent): Boolean {
         if (shouldUseCustomBackLongPress() && keyCode == KeyEvent.KEYCODE_BACK) {
             backLongPressJob?.cancel()
+
+            // check if the tab history if already visible.
+            if (event.eventTime - event.downTime >= ViewConfiguration.getLongPressTimeout() &&
+                TabHistoryDialogFragment().isOnTopOfBackQueue(navHost.navController)
```

Comment:
this line is creating a fragment instance just for a check using the class name. Would it be possible to refactor this as maybe instead an extension function of `NavController` where we just pass in a String to check against the destination name?

Reply:
I initially wanted to have a class as a receiver, but it had some downfalls that prevented getting the name in the extension method. I could try to refactor it for a `NavController` receiver, but it might sound kind of "unnatural" : `navcontroller.isOnTopOfBackQueue(fragmentName)`

Reply:
I might also have a String( the name) as a receiver, but it would not limit the instances that the extension method can use as a receiver that are not fragment names.

Reply:
Thanks for this. I updated it.